### PR TITLE
Node comms

### PR DESCRIPTION
**Pull request by andrew103**
Setup basic communications with generator node over pub/sub with a custom message type

### CODE REVIEW

File: trajectory_tracker/tracker.py [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
+from rclpy.node import Node
+
+from trajectory_interfaces.msg import Trajectory, SensorData
+
+class TrajectoryTracker(Node):
+
+  def __init__(self):
+    super().__init__('trajectory_generator')
+    self._publisher = self.create_publisher(SensorData, 'traj_start_data', 10)
+    self._subscription = self.create_subscription(Trajectory, 'new_traj', self.new_traj_callback, 10)
```
Comment (agupta231) [inline trajectory_tracker/tracker.py:11]: nit: more than 80 chars

File: trajectory_tracker/tracker.py [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+  trajectory_tracker = TrajectoryTracker()
+
+  rclpy.spin(trajectory_tracker)
+
+  rclpy.shutdown()
 
 
 if __name__ == '__main__':
-    main()
+  main()
```
Comment (agupta231) [inline trajectory_tracker/tracker.py:37]: Is there a reason why this is a function and not just in the if statement?

**Review by agupta231 — APPROVED** [review]
lgtm w/ small nits

File: trajectory_tracker/tracker.py (hunk 2 above)
Comment (andrew103) [reply trajectory_tracker/tracker.py:37]: Not really. It just got setup this way in autogeneration. I'll get rid of main

Comment (andrew103) [reply trajectory_tracker/tracker.py:37]: Jk having a main (or any function to start off the node) function is actually needed for ROS tracking

Comment (agupta231) [reply trajectory_tracker/tracker.py:44]: sgtm

**Review by agupta231 — APPROVED** [review]
lgtm

**Pull request merged**